src/frontend/http.rs

```diff
@@ -629,29 +724,45 @@ mod tests {
         );
     }
 
+    #[rstest]
     #[tokio::test]
-    async fn test_get_cached_write_query_error() {
-        let context = in_memory_context_with_single_table().await;
+    async fn test_get_cached_write_query_error(
+        #[values(None, Some("test_db"))] new_db: Option<&str>,
+    ) {
+        let context = in_memory_context_with_single_table(new_db).await;
         let handler = filters(context, http_config_from_access_policy(free_for_all()));
 
+        let mut uri = format!("/q/{}", str_to_hex_hash(CREATE_QUERY));
```

Comment:
You could factor these out into a tiny helper function `fn make_uri(query: &str, db: Option<&str>) -> String` to avoid some repetition.